packages/MonorepoBuilder/tests/VersionValidator/VersionValidatorTest.php

```diff
@@ -29,8 +32,8 @@ public function test(): void
         $this->assertArrayHasKey('some/package', $conflictingPackageVersionsPerFile);
 
         $expectedConflictingPackageVersionsPerFile = [
-            __DIR__ . '/Source/first.json' => '^1.0',
-            __DIR__ . '/Source/second.json' => '^2.0',
+            sprintf('%s%sSource%2$sfirst.json', __DIR__, DIRECTORY_SEPARATOR) => '^1.0',
+            sprintf('%s%sSource%2$ssecond.json', __DIR__, DIRECTORY_SEPARATOR) => '^2.0',
```

Comment:
Could you decouple this spritnf method to standalone line? Its too much operation goin on 1 line,making it difficult to read clearly

Reply:
Yeah I know, thats because the coding standards and phpstan always reported something new.
1. String concatenation with more that 4 parts => use sprintf (CS)
2. Using sprintf => use Safe\sprintf (phpstan)
3. Using Safe\sprintf => no longer readable

;)

Reply:
Fixed with 48465c2, both tools won't complain when using a second variable with fewer concatenations.

Reply:
> ...thats because the coding standards and phpstan always reported something new.

They're not perfect. 

But here it might makes sense too to create 2 tests, one for *nix and one for Windows. That would keep them readable and easier to maintain by particular os-devs.